src/server/common/helpers/next-page.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * @param {string} referrerURI
+ * @param {string} needle
+ */
+export default (referrerURI, needle) => {
+  try {
+    const { pathname } = new URL(referrerURI)
+    return {
+      nextPage: pathname === needle ? '/origin/summary' : ''
+    }
+  } catch (e) {
+    return {
+      nextPage: null
+    }
+  }
+}
```

Comment:
This approach feels fragile to me, as discussed on slack!


> the common alternative is using redirect_uri query param - so the page state depends on session state & url
> in the referrer version, it's session state, url & the referrer (which doesn't survive copy & pasting for example)

This is the one thing I think we should change, unless there're pros to the referrer version that I'm missing.